src/serac/physics/utilities/variational_form/tuple_arithmetic.hpp

```diff
@@ -0,0 +1,214 @@
+#pragma once
+
+#include <tuple>
+
+#include "tensor.hpp"
+
+namespace serac {
+
+namespace detail {
+
+/**
+ * @brief Trait for checking if a type is a @p std::tuple
+ */
+template <typename T>
+struct is_tuple {
+  // FIXME: Should we use std::false_type/std::true_type
+  static constexpr bool value = false;
+};
+/// @overload
+template <typename... T>
+struct is_tuple<std::tuple<T...> > {
+  static constexpr bool value = true;
+};
+
+/**
+ * @brief Trait for checking if a type if a @p std::tuple containting only @p std::tuple
+ */
+template <typename T>
+struct is_tuple_of_tuples {
+  static constexpr bool value = false;
+};
+/// @overload
+template <typename... T>
+struct is_tuple_of_tuples<std::tuple<T...> > {
+  static constexpr bool value = (is_tuple<T>::value && ...);
+};
+
+/////////////////////////////////////////////////
+
+template <typename... S, typename... T, int... I>
+constexpr auto plus_impl(const std::tuple<S...>& A, const std::tuple<T...>& B, std::integer_sequence<int, I...>)
+{
+  return std::make_tuple((std::get<I>(A) + std::get<I>(B))...);
+}
+
+template <typename... S, typename... T, int... I>
+constexpr auto minus_impl(const std::tuple<S...>& A, const std::tuple<T...>& B, std::integer_sequence<int, I...>)
+{
+  return std::make_tuple((std::get<I>(A) - std::get<I>(B))...);
+}
+
+template <typename... S, typename T, int... I>
+constexpr auto mult_impl(const std::tuple<S...>& A, T scale, std::integer_sequence<int, I...>)
+{
+  return std::make_tuple(std::get<I>(A) * scale...);
+}
+
+template <typename S, typename... T, int... I>
+constexpr auto mult_impl(S scale, const std::tuple<T...>& A, std::integer_sequence<int, I...>)
+{
+  return std::make_tuple((scale * std::get<I>(A))...);
+}
+
+template <int I, int... i>
```

Comment:
Upper and lower case variables in the same function are prone to errors. I would prefer more verbose names.